src/main/java/org/wildfly/security/credential/store/impl/KeyStoreCredentialStore.java

```diff
@@ -655,6 +655,7 @@ public void remove(final String credentialAlias, final Class<? extends Credentia
                     }
                 }
             }
+            cache.remove(credentialAlias);
```

Comment:
KeyStoreException is thrown mainly in case the entry cannot be removed from key store. In that case I prefer to have alias still in cache.